core/templates/dev/head/player/AnswerClassificationService.js

```diff
@@ -97,13 +97,23 @@ oppia.factory('answerClassificationService', [
           explorationId, oldState, answer, isInEditorMode,
           interactionRulesService) {
         var deferred = $q.defer();
-        if (interactionRulesService) {
-          var answerGroups = oldState.interaction.answer_groups;
-          var defaultOutcome = oldState.interaction.default_outcome;
+        var result = null;
+        var answerGroups = oldState.interaction.answer_groups;
+        var defaultOutcome = oldState.interaction.default_outcome;
 
-          deferred.resolve(classifyAnswer(
-            answer, answerGroups, defaultOutcome, interactionRulesService));
+        if (interactionRulesService) {
+          result = classifyAnswer(
+            answer, answerGroups, defaultOutcome, interactionRulesService);
         } else {
+          alertsService.addWarning(
+            'Something went wrong with the exploration.');
```

Comment:
Can this error be a little more explicit, like: "Something went wrong with the exploration: no interactionRulesService provided" (or something like that)? Otherwise, debugging things in the future is going to be trickier, since you have this same message in multiple locations.